app/src/main/java/com/rkm/tasky/repository/abstraction/TaskyReminderRepository.kt

```diff
@@ -0,0 +1,63 @@
+package com.rkm.tasky.repository.abstraction
+
+import com.rkm.tasky.database.model.ReminderEntity
+import com.rkm.tasky.network.model.Reminder
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.Result
+import java.util.UUID
+
+interface TaskyReminderRepository {
+    suspend fun getReminder(id: String): Result<ReminderDTO?, NetworkError.APIError>
+    suspend fun createReminder(reminder: ReminderDTO): EmptyResult<NetworkError.APIError>
+    suspend fun updateReminder(reminder: ReminderDTO): EmptyResult<NetworkError.APIError>
+    suspend fun deleteReminder(reminder: ReminderDTO): EmptyResult<NetworkError.APIError>
```

Comment:
Repository functions should work with domain models and its implementation should deal with the data layer models